modules/core/src/v2/coins/algo.ts

```diff
@@ -151,9 +151,10 @@ export class Algo extends BaseCoin {
     if (!keys.prv) {
       throw new Error('Missing prv in key generation.');
     }
+
     return {
-      pub: keys.pub,
-      prv: keys.prv,
+      pub: keyPair.getAddress(),
```

Comment:
These methods shall go to base keypair class. Approving it now for demo. @rliberal-simtlix Please track and create a separate ticket to fix this,